operators/tokenizer/bpe_tokenizer.hpp

```diff
@@ -269,9 +274,14 @@ class VocabData {
     return special_tokens_.SplitBySpecialTokens(input);
   }
 
-  int GetEncoding(const std::string& key) {
+  // Returns {true, token} if key was found in vocab, and {false, -1} otherwise
+  std::pair<bool, int> GetEncoding(const std::string& key) {
     auto it = vocab_map_.find(key);
-    return it->second;
```

Comment:
There is already a unknown token id in this class, GetEncoding can return unknown_id if not found, instead of std::pair.